src/common/tusb_types.h

```diff
@@ -342,9 +350,14 @@ typedef struct TU_ATTR_PACKED
   } bmAttributes     ; ///< This field describes the endpoint's attributes when it is configured using the bConfigurationValue. \n Bits 1..0: Transfer Type \n- 00 = Control \n- 01 = Isochronous \n- 10 = Bulk \n- 11 = Interrupt \n If not an isochronous endpoint, bits 5..2 are reserved and must be set to zero. If isochronous, they are defined as follows: \n Bits 3..2: Synchronization Type \n- 00 = No Synchronization \n- 01 = Asynchronous \n- 10 = Adaptive \n- 11 = Synchronous \n Bits 5..4: Usage Type \n- 00 = Data endpoint \n- 01 = Feedback endpoint \n- 10 = Implicit feedback Data endpoint \n- 11 = Reserved \n Refer to Chapter 5 of USB 2.0 specification for more information. \n All other bits are reserved and must be reset to zero. Reserved bits must be ignored by the host.
 
   struct TU_ATTR_PACKED {
+#if defined(__CCRX__)
+    //FIXME the original defined bit field has a problem with the CCRX toolchain, so only a size field is defined
+    uint16_t size;
+#else
```

Comment:
I am still not convinced there is issue with bit-field within a struct. The bitfield order should give us the correct layout. Maybe it is endian issue. We could just enable the LOG=2 to see if the size of openning endpoit is correct or not. If you are busy, maybe we could revert this for now to get PR to merge first (I could do the revert).

Reply:
@Wini-Buh is it OK if I revert this change for now to get this PR merged first. I am ok for this change in future PR, I only need confirmation that it does indeed needed for ccrx to work.

Reply:
For me it's OK to revert it. It is definitively required in the project I'm using tinyUSB, which is build in `big endian` and `left bitfield order`. I have already setup test code to investigate the problem deeper (for me it looks like a problem of the CCRX toolchain if the configuration I mentioned is used... and if so, there is maybe another workaround for it). 
I keep you informed about the result of the investigation about this problem.

Reply:
Thanks for confirmation, I will leave it on for now since you actually need it for ccrx. We could update it later as follow up PR. However, the bmAttributes (just right above) also use bitfield within struct, you may also need to check it out as well. Too bad, I couldn't test ccrx myself, they make it difficult to run on Linux.

Anyway, when you have time please try to run an very simply example using tusb_desc_endpoint_t struct (copy it to helloworld example), then write an value to the wMaxPacketSize like 0x1234 then print out its field out to see how it is layout etc ... I tried to run it within my Windows VM, but got issue connecting to E2 debugger `GDB action 'connect to target', has failed with error code, 0xffffffffd`. I just gave up on trying it out for now.